dipy/viz/horizon/visualizer/slice.py

```diff
@@ -36,8 +35,11 @@ def __init__(
 
         self._slice_actors = [None] * 3
 
-        self._data_ndim = data.ndim
-        self._data_shape = data.shape
+        if len(self._data.shape) == 4 and self._data.shape[-1]:
```

Comment:
Is there a reason we check for self._data.shape[-1]? Are there cases we get a shape something like (256, 256, 256, None) ?

Reply:
No that is a typo we are just checking for the volumes with single channel. Where we convert it to a normal 3D image instead of 4D image.